src/electionguard/election.cpp

```diff
@@ -37,13 +38,27 @@ namespace electionguard
         unique_ptr<ElementModQ> manifestHash;
         unique_ptr<ElementModQ> cryptoBaseHash;
         unique_ptr<ElementModQ> cryptoExtendedBaseHash;
+        unordered_map<string, string> extendedData;
 
         Impl(uint64_t numberOfGuardians, uint64_t quorum, unique_ptr<ElementModP> elGamalPublicKey,
              unique_ptr<ElementModQ> commitmentHash, unique_ptr<ElementModQ> manifestHash,
              unique_ptr<ElementModQ> cryptoBaseHash, unique_ptr<ElementModQ> cryptoExtendedBaseHash)
             : elGamalPublicKey(move(elGamalPublicKey)), commitmentHash(move(commitmentHash)),
               manifestHash(move(manifestHash)), cryptoBaseHash(move(cryptoBaseHash)),
               cryptoExtendedBaseHash(move(cryptoExtendedBaseHash))
+        {
+            this->numberOfGuardians = numberOfGuardians;
+            this->quorum = quorum;
+            this->extendedData = {};
+        }
+
+        Impl(uint64_t numberOfGuardians, uint64_t quorum, unique_ptr<ElementModP> elGamalPublicKey,
```

Comment:
this pointer to implementation pattern allows the c++ specification in the header to remain independent from the actual implementation of the data structure. each class type has an `Impl` which is again an opaque structure externally.